.github/workflows/release-workflow.yml

```diff
@@ -0,0 +1,45 @@
+name: E4E Release & Publish Workflow
+on:
+  pull_request:
+    types: [closed]
+    branches:
+      - master
+if: github.event.pull_request.merged
+env:
+  PROJECT_NAME: explorer-for-endevor
+jobs:
+  setup:
+    runs-on: windows-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+      - name: Get E4E package version
+        uses: nyaa8/package-version@v1.0.3
+        id: npm-version
+        env:
+          ENDEVOR_VERSION_PATH: packages/${{ env.PROJECT_NAME }}/package.json
+        with:
+          path: ${{ env.ENDEVOR_VERSION_PATH }}
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+
+  release-and-publish:
+    runs-on: windows-latest
+    steps:
+      - name: Publish to Open VSX Registry
+        uses: HaaLeo/publish-vscode-extension@v0
+        id: publishToOpenVSX
+        env:
+          VSIX_PATH: packages/${{ env.PROJECT_NAME }}/vsce/*.vsix
```

Comment:
Hmm, but if it is a separate workflow and a separate job, how the packaged vsix file will be in that folder?

GH actions will open the same workspace here after the build workflow? Did you specify it somewhere?

As far as I understand, all the GH actions workflows and jobs are starting in a fresh Docker container with an empty workspace until user specified, that they need to reuse the same workspace from the previous workflow. I don't think, that reusing workspaces is a good idea, because we will have to handle state problems. And we will end up like Jenkins :) I love Jenkins, but I don't want to repeat their mistakes.

Maybe we can repeat all required build and package steps for the master branch in this workflow to get a ready for release vsix file? Also we can specify to run the `CI` pipeline for all the branches, except the master branch.

Reply:
What if I just move the `create-artifact` step from the CI pipeline to the Packaging pipeline? We won't get an artifact after running tests, but we don't get one now anyway...what do you think?

Reply:
Well, we still need to have a working artifact even in the feature branches (for validation purposes), so I think, the easiest approach will be to duplicate this job for both releasing and CI workflows. Will this work?